bgpd/bgp_pbr.c

```diff
@@ -1841,6 +1843,8 @@ static void bgp_pbr_policyroute_add_to_zebra_unit(struct bgp *bgp,
 	temp2.proto = bpf->protocol;
 	bpme = hash_get(bpm->entry_hash, &temp2,
 			bgp_pbr_match_entry_alloc_intern);
+	assert(bpme);
```

Comment:
This can never be null, why would we need an assert here?

Reply:
Hi. I'm not sure, but I guess Coverity would complain, too, in that case, because in hash_get() there are code paths involving returning NULL, e.g.
```
        if (alloc_func) {                                                       
                newdata = (*alloc_func)(data);                                  
                if (newdata == NULL)                                            
                        return NULL;                                            
```

Reply:
Because there were NULL checks for bmp and bmpe, so Coverity could use that as a tip. We could remove the checks and see what happens :-)

Reply:
(in the sense of checking if Coverity was complaining because of the NULL check and not because of an inductive proof)

Reply:
@qlyoung I'll do that, thank you! :-)